### PR TITLE
Move LogicalObjectFifoType to tablegen

### PR DESCRIPTION
**Pull request by jtuyls**
Address https://github.com/nod-ai/iree-amd-aie/pull/267/files#r1563206551 and https://github.com/nod-ai/iree-amd-aie/pull/267/files#r1563205156

### CODE REVIEW

**Review by Abhishek-Varma — APPROVED** [review]
One small change. LGTM otherwise!

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEDialect.cpp [hunk 1]
```diff
@@ -8,11 +8,9 @@
 
 #include "iree-amd-aie/IR/AMDAIEAttrs.h"
 #include "iree-amd-aie/IR/AMDAIEDialect.cpp.inc"
```
[the remark this answers is not in the document]
Comment (jtuyls) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEDialect.cpp:10]: I get this error: `error: undefined symbol: mlir::detail::TypeIDResolver<mlir::iree_compiler::AMDAIE::AMDAIEDialect, void>::id`

[the remark this answers is not in the document]
Comment (jtuyls) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEDialect.cpp:10]: And I did not add that line in my previous PR, it was already there: https://github.com/nod-ai/iree-amd-aie/blame/7d383d94011cd3e2c012661c9bbe87fedb67f15d/compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEDialect.cpp#L10. If it shouldn't be there, I think this is out of scope of this PR and should be handled in another one.

**Review by yzhang93 — APPROVED** [review]

**Review by MaheshRavishankar — COMMENTED** [review]
Nicely done!

**Pull request merged**